### PR TITLE
Mypy

### PR DESCRIPTION
**Pull request by tpazderka**
- [ ] Any changes relevant to users are recorded in the `CHANGELOG.md`.
- [ ] The documentation has been updated, if necessary.
---

### CODE REVIEW

**Comment by tpazderka** [comment]
Hm, the Py34 on Windows is failing for some reason. I am considering dropping the support for Py34 as its EOL is in about a month...

@schlenk Any objections?

**Comment by schlenk** [comment]
@tpazderka I don't have objections about dropping Python 3.4 alongside the Python 2.7 drop, its basically the same reasoning.

**Comment by tpazderka** [comment]
OK, 3.4 removed as well.

File: src/oic/oic/provider.py [hunk 1]
```diff
@@ -247,9 +247,12 @@ def __init__(self, name, sdb, cdb, authn_broker, userinfo, authz,
         self.extra_claims = extra_claims
         self.extra_scope_dict = extra_scope_dict
 
+        self.register_endpoint = None
         for endp in self.endp:
             if endp.etype == 'registration':
                 endpoint = urljoin(self.baseurl, endp.url)
+                DeprecationWarning("Using `register_endpoint` is deprecated, please use "
```
Comment (schlenk) [inline src/oic/oic/provider.py]: Why only create a `DeprecationWarning` instance? Shouldn't it be used with warning.warn(...) ?

File: src/oic/utils/http_util.py [hunk 2]
```diff
@@ -546,14 +546,15 @@ def wsgi_wrapper(environ, start_response, func, **kwargs):
...
 
-    def setServer(self, server):
+    @srv.setter
+    def srv(self, server):
         self._srv = server
 
-    srv = property(getServer, setServer)
-
     def __init__(self, srv, ttl=5, secure=True, httponly=True):
         self.srv = None
```
Comment (schlenk) [inline src/oic/utils/http_util.py]: Why keep this when having the property above?

File: src/oic/oic/provider.py (hunk 1 above)
Comment (tpazderka) [reply src/oic/oic/provider.py]: Yeah, dumb mistake :)

File: src/oic/utils/http_util.py (hunk 2 above)
Comment (tpazderka) [reply src/oic/utils/http_util.py]: True

**Pull request merged**